go-controller/pkg/ovn/gateway_init.go

```diff
@@ -316,13 +317,38 @@ func addDistributedGWPort() error {
 		return fmt.Errorf("failed to get master's chassis ID error: %v", err)
 	}
 
+	var cidrs []string
+	var macStr string
+	var natSubnetNextHop string
+	if config.IPv4Mode {
+		dgpIP := net.ParseIP(util.V4NodeLocalDistributedGwPortIP)
+		cidrs = append(cidrs,
+			fmt.Sprintf("%s/%d", dgpIP.String(), util.V4NodeLocalNatSubnetPrefix))
+		macStr = util.IPAddrToHWAddr(dgpIP).String()
+		natSubnetNextHop = util.V4NodeLocalNatSubnetNextHop
+	}
+	if config.IPv6Mode {
+		dgpIP := net.ParseIP(util.V6NodeLocalDistributedGwPortIP)
```

Comment:
There needs to be some sort of loop here for dual-stack, to configure both the ipv4 and ipv6 versions